reana_commons/snakemake.py

```diff
@@ -9,30 +9,41 @@
 """REANA Snakemake Workflow utils."""
 
 import os
+from itertools import filterfalse, chain
+from typing import Any, Dict, List, Optional
+
 from snakemake import snakemake
+from snakemake.dag import DAG
 from snakemake.io import load_configfile
+from snakemake.jobs import Job
+from snakemake.persistence import Persistence
+from snakemake.rules import Rule
 from snakemake.workflow import Workflow
 
 from reana_commons.errors import REANAValidationError
 
 
-def snakemake_load(workflow_file, **kwargs):
+def snakemake_load(workflow_file: str, **kwargs: Any) -> Dict:
     """Validate Snakemake workflow specification.
 
     :param workflow_file: A specification file compliant with
         `snakemake` workflow specification.
     :type workflow_file: string
 
-    :returns: Empty dict as the workflow spec remains in the Snakefile.
+    :returns: Dictonary containing relevant workflow metadata.
     """
 
-    def _create_snakemake_workflow(snakefile, configfiles=None):
-        """Create ``snakemake.workflow.Workflow`` instance.
+    def _create_snakemake_dag(
+        snakefile: str, configfiles: Optional[List[str]] = None, **kwargs: Any
+    ) -> DAG:
```

Comment:
*suggestion*: This is a nested function inside `snakemake_load`. It doesn't look like it is used more than once in `snakemake_load` and it cannot be used outside so the purpose of it is not clear. I would suggest or move it outside on the same level as `snakemake_load` or remove the function and move the code to the `snakemake_load` body.

Reply:
>  It doesn't look like it is used more than once in snakemake_load and it cannot be used outside

That's precisely the reason why it's nested into `snakemake_load`.

Since this function partially replicates part of snakemake functionality, I prefer to keep this code separated from _our code_. I think this way it'd be easier to introduce changes in the future if needed.